src/Application/Factories/AlarmSubscriptionProviderFactory.cs

```diff
@@ -0,0 +1,79 @@
+﻿// SPDX-License-Identifier: MIT. 
+// Copyright Contributors to the Open Manufacturing Platform.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OMP.Connector.Application.Providers.AlarmSubscription;
+using OMP.Connector.Application.Validators;
+using OMP.Connector.Domain.Configuration;
+using OMP.Connector.Domain.Models.Telemetry;
+using OMP.Connector.Domain.OpcUa;
+using OMP.Connector.Domain.Providers;
+using OMP.Connector.Domain.Schema.Interfaces;
+using OMP.Connector.Domain.Schema.Request.AlarmSubscription;
+
+namespace OMP.Connector.Application.Factories
+{
+    public class AlarmSubscriptionProviderFactory : IAlarmSubscriptionProviderFactory
+    {
+        private readonly IAlarmSubscriptionRepository _subscriptionRepository;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IOptions<ConnectorConfiguration> _connectorConfiguration;
+        private readonly AlarmMonitoredItemServiceInitializerFactoryDelegate _monitoredItemServiceInitializerFactory;
+        private readonly AlarmMonitoredItemValidator _alarmMonitoredItemValidator;
+
+        public AlarmSubscriptionProviderFactory(
+            IAlarmSubscriptionRepository dataManagementService,
+            ILoggerFactory loggerFactory,
+            IOptions<ConnectorConfiguration> connectorConfiguration,
+            AlarmMonitoredItemServiceInitializerFactoryDelegate monitoredItemServiceInitializerFactory,
+            AlarmMonitoredItemValidator alarmMonitoredItemValidator)
```

Comment:
Can this not be an interface? Do we do it as instances in other areas? (Not sure if already broke this rule elsewhere)

Reply:
Or the AbstractValidator<..> to minimize the coupling and ease testing?

Reply:
Although the method was removed as part of another commit based on another code review comment, I did implement the abstraction on field declarations, constructors and some other method signatures.